models/views/users_table_view_permissions.sql

```diff
@@ -0,0 +1,46 @@
+with tables as (
+
+  select * from {{ref('pg_tables')}}
+
+), views as (
+
+  select * from {{ref('pg_views')}}
+
+), users as (
+
+  select * from {{ref('pg_user')}}
+
+), objects as (
+  
+  select
+    schema_name
+  , 'table' AS object_type
+  , table_name AS object_name
+  , schema_name + '.' + table_name AS full_object_name
```

Comment:
i had a small issue here when running this on my Redshift cluster -- I had a table named `data.table`, so this line generated a `full_object_name` of:

```
schema.data.table
```

The `has_table_privilege` function down below got confused and thought that I was querying for `"schema"."data"."table"`

Unsure how common this would be, but I think you can avoid this error by quoting the schema / table names, eg:

```
'"' || schema_name || '"."' table_name || '"' AS full_object_name
```

_that_ would probably break if your table name somehow has a quote in it though :)

Reply:
I'm more interested in how you managed to put a `.` in a table name! The [docs](https://docs.aws.amazon.com/redshift/latest/dg/r_names.html) suggest that wouldn't be possible, and I tried creating a table with a `.` only to get an error.

Reply:
```
create table "use""quotes"" as (select 1 as id);
create table ""quotes"" as (select 1 as id);
```
The above both throw errors for me, so I suspect that quoted identifiers are not allowed as table names, in which case I should implement your suggestion. Good catch!